activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -66,7 +66,7 @@ def require_counter_update?
 
         def update_counters_on_replace(record)
           if require_counter_update? && different_target?(record)
-            owner.instance_variable_set :@_after_replace_counter_called, true
+            owner.instance_variable_set :@_after_replace_counter_called, { reflection.foreign_key => true }
```

Comment:
You are not considering previous value of hash, It will be better to add / merge key, value pair

Reply:
Yes you are right, I will fix it soon